util: clean up retry

andrewbaptist opened the pull request:
Previously there were two retry.Start methods one that took a context and one that didn't. There have been several bugs related to calling the one without the context. This commit requires a context is passed to Start and removes the other version.

Epic: None

Release note: None

Commit:
util: clean up retry

Previously there were two retry.Start methods one that took a context
and one that didn't. There have been several bugs related to calling
the one without the context. This commit requires a context is passed to
Start and removes the other version.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/124924)
<!-- Reviewable:end -->

Commit:
util: remove redundant retry multiplier

Previously a lot of code was manually setting the retry multiplier. In
most cases it was setting it to the default value of 2, but in some
cases it was using lower (or higher) values. In general exponential
backoff with a value of 2 works best in almost all cases. This PR
cleans up all those places.

Release note: None

Commit:
util: remove base.DefaultRetryOptions

Previously the method `DefaultRetryOptions` constructed a
`retry.Options` that was slightly different than the one constructed
directly using `retry.Options{}`. There doesn't appear to be any
advantage of using one vs the other so by cleaning this up it will add
consistency to the code.

Epic: none

Release note: None
